lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,327 @@
+    setState(() {
+      filteredUsers = store.users.values
+        .where((user) =>
+          user.userId != store.selfUserId &&
+          user.fullName.toLowerCase().contains(searchController.text.toLowerCase()))
+        .toList()
+        ..sort((a, b) {
+          final recentDms = store.recentDmConversationsView;
+          final aLatestMessageId = recentDms.latestMessagesByRecipient[a.userId];
+          final bLatestMessageId = recentDms.latestMessagesByRecipient[b.userId];
+
+          if (aLatestMessageId != null && bLatestMessageId != null) {
+            return bLatestMessageId.compareTo(aLatestMessageId);
+          }
+          if (aLatestMessageId != null) return -1;
+          if (bLatestMessageId != null) return 1;
+          return 0;
+        });
+    });
+  }
+
+  // Scroll to the search field when the user selects a user
+  void scrollToSearchField() {
+    SchedulerBinding.instance.addPostFrameCallback((_) {
+      if (scrollController.hasClients) {
+        scrollController.animateTo(
+          scrollController.position.maxScrollExtent,
+          duration: const Duration(milliseconds: 200),
+          curve: Curves.easeOut);
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: MediaQuery.of(context).size.height * 0.95,
+      child: Column(children: [
+        NewDmHeader(
+          selectedUserIds: selectedUserIds,
+          onClose: () => Navigator.pop(context)),
+        NewDmSearchBar(
+          searchController: searchController,
+          scrollController: scrollController,
+          selectedUserIds: selectedUserIds),
+        Expanded(child: NewDmUserList(
+          filteredUsers: filteredUsers,
+          selectedUserIds: selectedUserIds,
+          onUserSelected: (userId) {
+            setState(() {
+              if (selectedUserIds.contains(userId)) {
+                selectedUserIds.remove(userId);
+              } else {
+                selectedUserIds.add(userId);
+              }
+              scrollToSearchField();
+            });
+          }))]));
+  }
+}
+
+class NewDmHeader extends StatelessWidget {
+  const NewDmHeader({
+    super.key,
+    required this.selectedUserIds,
+    required this.onClose});
+
+  final Set<int> selectedUserIds;
+  final VoidCallback onClose;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    return Container(
+      padding: const EdgeInsets.all(8),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          _buildBackButton(context, designVariables, zulipLocalizations),
+          Text(zulipLocalizations.newDmSheetScreenTitle,
+            style: TextStyle(color: designVariables.title,fontSize: 20, fontWeight: FontWeight.w600)),
```

Comment:
nit:
```suggestion
style: TextStyle(color: designVariables.title, fontSize: 20, fontWeight: FontWeight.w600)),
```

Reply:
To control font weights, we usually use `weightVariableTextStyle`. See its dartdoc and existing places that use it for reference.

Reply:
The design also specifies the line-height for this element.

![image](https://github.com/user-attachments/assets/1ee9fc72-c462-447c-b581-148e5503387c)

The Flutter equivalent of this is `height`/`fontSize`. You can find some examples like:

```dart
      child: Text(label(zulipLocalizations),
        style: const TextStyle(fontSize: 20, height: 24 / 20)
          .merge(weightVariableTextStyle(context, wght: 600)),
```